### DISCUSSION TITLE
elementary OS mobile

### DISCUSSION THREAD

**Discussion by SubhadeepJasu** [elementary/triage#138] [Ideas]
Now that it's up in the air, I am posting some concepts that I have worked on long time ago. From a technical standpoint, I imagine that there can be a modification to gala to make it work in a "phone" form factor and interface. Granite can be updated to add support for and encourage responsive design and the concept of convergence.

**Demo**
https://www.figma.com/proto/BrOcTkCRaD5hTKuOQnvM7L/elementary-OS-Mobile?node-id=0%3A1

**Source**
https://www.figma.com/file/BrOcTkCRaD5hTKuOQnvM7L/elementary-OS-Mobile

**Screenshots**
<p align="center">
<img width="500" alt="Home Screen" src="https://user-images.githubusercontent.com/20795161/160253859-0a66d5c2-0029-4d96-9f7f-f2af18fbe192.png"><br>
<img width="200" alt="CommandBar LockScreen" src="https://user-images.githubusercontent.com/20795161/160253869-ebcb316b-6cc8-4b65-bbbc-b662297162bb.png">&nbsp;<img width="200" alt="Pebbles (Google Pixel 2 XLSize) 1" src="https://user-images.githubusercontent.com/20795161/160253875-a25638d6-ab13-4525-b0e3-5483a2c539c1.png">&nbsp;<img width="200" alt="Phone Flow Main" src="https://user-images.githubusercontent.com/20795161/160253879-8ccdd626-46ed-46ef-b7d7-341db5cd5ff2.png">&nbsp;<img width="200" alt="Multitasking View" src="https://user-images.githubusercontent.com/20795161/160253883-c8591375-bd0d-45ef-a0dd-85964cd1a4a1.png">
</p>

**Comment by zeebok** [comment]
This looks great! Based on discussions on Twitter this is definitely a good time to share this. While I don't know a huge amount for mobile/responsive app design and what is available already in GTK, I feel like as a dev some way to check if it is a mobile or desktop mode would be a must. I also think AppCenter would need to be able to inform a user what form factors an app supports.

In general this makes me hope for eOS on mobile and I love how it looks already!

**Reply by SubhadeepJasu** [reply]
I agree. Responsive design can be a part of the design guideline, but a developer can still publish apps that only look good in desktop mode. One way to tackle this is to have minimum window size be specified in the app metadata. Based on this AppCenter can show a warning for apps that don't fit the screen of the device (either in portrait or landscape mode). Gala can attempt to render any app, and also force the main window to be a certain resolution when running on phone.

**Comment by ChildishGiant** [comment]
Something about these mockups remind me of luna :thinking: Maybe the rounding on buttons and the colour of the command bar buttons

**Reply by SubhadeepJasu** [reply]
It could be. I was looking at earlier elementary OS concepts for reference.

**Comment by SubhadeepJasu** [comment]
If anyone (ux designers or os devs) needs access to the figma file for editing purpose, just let me know.

**Reply by cosmic-zip** [reply]
Finding a physical device to properly test apps seems to be more of a challenge.

**Comment by IdkWhatToCallMe123** [comment]
maybe a bit late but the clock on the statusbar should definitely not be centered considering that pretty much all phones have a notch or a holepunch there
